server/etcdserver/api/membership/cluster.go

```diff
@@ -303,9 +295,16 @@ func (c *RaftCluster) Recover(onSet func(*zap.Logger, *semver.Version)) {
 
 // ValidateConfigurationChange takes a proposed ConfChange and
 // ensures that it is still valid.
-func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
-	// TODO: this must be switched to backend as well.
-	membersMap, removedMap := membersFromStore(c.lg, c.v2store)
+func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange, shouldApplyV3 ShouldApplyV3) error {
+
+	var membersMap map[types.ID]*Member
+	var removedMap map[types.ID]bool
+	membersMap, removedMap = c.be.MustReadMembersFromBackend()
+	if !shouldApplyV3 {
+		//TODO geetasg - RaftCluster purely based on backend cannot validate an entry older than ci
```

Comment:
What do you mean by validate?

Reply:
I had a test failure on validation which looked like was due to trying to validate using 'future' info. Ref:https://github.com/etcd-io/etcd/pull/15905/files#diff-164b95d19232df0c51cd511fe4ca61de738660fc11db6ddcdafa0ec5f5c80a35R1978

I will revisit to add a test for that code path once it is better understood.